tests/devices/test_default_mixed.py

```diff
@@ -427,6 +427,118 @@ def test_diag_root(self, x):
         assert np.allclose(dev._state, target_state)
 
 
+class TestApplyBasisState:
+    """Unit tests for the method `_apply_basis_state"""
+
+    @pytest.mark.parametrize("nr_wires", [1, 2, 3])
+    def test_all_ones(self, nr_wires):
+        """Tests that the state |11...1> is applied correctly"""
+        dev = qml.device("default.mixed", wires=nr_wires)
+        state = np.ones(nr_wires)
+        dev._apply_basis_state(state, wires=Wires(range(nr_wires)))
+        b_state = basis_state(2 ** nr_wires - 1, nr_wires)
+        target_state = np.reshape(b_state, [2] * 2 * nr_wires)
```

Comment:
So, `basis_state()` is an internal function I created within the tests files. It's just 
```python3
def basis_state(index, nr_wires):
    rho = np.zeros((2 ** nr_wires, 2 ** nr_wires), dtype=np.complex128)
    rho[index, index] = 1
    return rho
```
Since specific states are used *a lot* throughout the tests, I found it best to define these very simple functions, whose correctness is almost self-evident. 

Do you still think it's worth making the tests even more explicit?

Reply:
Personally, I would favour duplicating the two lines

```python
    rho = np.zeros((2 ** nr_wires, 2 ** nr_wires), dtype=np.complex128)
    rho[index, index] = 1
```

within the tests (but curious to get others thoughts as well)

Reply:
I did a quick count and there are a total of 41 tests that would need to be changed this way. I can do it, but is it worth it?

Reply:
No, this change isn't required for my approval; instead it is more of a 'nitpick' or personal preference as to how I would liked to have read/written the tests.

So up to you! Depends how convinced you are by my arguments 🙂